app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/ContactDetailsConfirmAddressController.scala

```diff
@@ -50,26 +52,51 @@ class ContactDetailsConfirmAddressController @Inject() (
 )(implicit ec: ExecutionContext)
     extends FrontendController(mcc) with Cacheable with I18nSupport {
 
+  //TODO - temporary while we're working out where to get it from
+  val updateSoleTraderAddress
+    : (String, JourneyRequest[AnyContent], HeaderCarrier) => Future[Address] = (_, request, _) =>
+    updateBusinessAddress(
+      Address(addressLine1 = "2 Scala Street",
+              addressLine2 = Some("Soho"),
+              townOrCity = "London",
+              postCode = "W1T 2HN"
+      )
+    )(request)
+
+  val updateUkCompanyAddress
+    : (String, JourneyRequest[AnyContent], HeaderCarrier) => Future[Address] =
+    (journeyId, request, hc) =>
+      incorpIdConnector.getDetails(journeyId)(hc)
+        .flatMap(response => updateBusinessAddress(response.companyAddress.toPptAddress)(request))
+
```

Comment:
I am not sure a closure provides tangible benefit here. In this case you cannot have `request` and `hr` as **implicits** and you need to explicitly call them out on the subsequent calls. 

All the benefits we get from Scala implicits are eroded by the closure. 

I can keep it the approach. Once we have more legal entities to support, potentially we can discuss the design again.

Reply:
yeap, I inlined those methods now and using implicits